source/DasBlog.Web.UI/Startup.cs

```diff
@@ -1,115 +1,155 @@
-            }
-
-            app.UseStaticFiles();
-
-            app.UseMvc(routes =>
-            {
-                routes.MapRoute(
-                    "Original Post Format",
-                    "{posttitle}.aspx",                           
-                    new { controller = "Home", action = "Post", posttitle = "" });
-
-                routes.MapRoute(
-                    "New Post Format",
-                    "{posttitle}",
-                    new { controller = "Home", action = "Post", id = "" });
-
-                routes.MapRoute(
-                    name: "default",
-                    template: "{controller=Home}/{action=Index}/{id?}");
-            });
-
-            RewriteOptions options = new RewriteOptions()
-                 .AddIISUrlRewrite(env.ContentRootFileProvider, @"Config\IISUrlRewrite.xml");
-
-            app.UseRewriter(options);
-        }
-    }
+	public class Startup
+	{
+		private IHostingEnvironment _hostingEnvironment;
+
+		public Startup(IConfiguration configuration, IHostingEnvironment env)
+		{
+			var builder = new ConfigurationBuilder()
+			.SetBasePath(env.ContentRootPath)
+			.AddXmlFile(@"Config\site.config", optional: true, reloadOnChange: true)
+			.AddXmlFile(@"Config\metaConfig.xml", optional: true, reloadOnChange: true)
+			.AddXmlFile(@"Config\siteSecurity.config", optional: true, reloadOnChange: true)
+			.AddJsonFile("appsettings.json", optional: true, reloadOnChange: true)
+			.AddJsonFile($"appsettings.{env.EnvironmentName}.json", optional: true)
+			.AddEnvironmentVariables();
+
+			Configuration = builder.Build();
+
+			_hostingEnvironment = env;
+		}
+
+		public IConfiguration Configuration { get; }
+
+		// This method gets called by the runtime. Use this method to add services to the container.
+		public void ConfigureServices(IServiceCollection services)
+		{
+			services.AddOptions();
+			services.AddMemoryCache();
+
+			services.Configure<SiteConfig>(Configuration);
+			services.Configure<MetaTags>(Configuration);
+			services.Configure<SiteSecurityConfig>(Configuration);
+
+			// Add identity types
+			services
+				.AddIdentity<DasBlogUser, DasBlogRole>()
+				.AddDefaultTokenProviders();
+
+			services.Configure<IdentityOptions>(options =>
+			{
+				// Password settings
+				options.Password.RequireDigit = true;
+				options.Password.RequiredLength = 8;
+				options.Password.RequireNonAlphanumeric = false;
+				options.Password.RequireUppercase = true;
+				options.Password.RequireLowercase = false;
+				options.Password.RequiredUniqueChars = 6;
+
+				// Lockout settings
+				options.Lockout.DefaultLockoutTimeSpan = TimeSpan.FromMinutes(30);
+				options.Lockout.MaxFailedAccessAttempts = 10;
+				options.Lockout.AllowedForNewUsers = true;
+
+				// User settings
+				options.User.RequireUniqueEmail = true;
+			});
+
+			services.ConfigureApplicationCookie(options =>
+			{
+				// Cookie settings
+				options.Cookie.HttpOnly = true;
```

Comment:
This is why I love others getting involved, I learn so much. I only knew about how this kind of info is configure in a full ASP.NET context with a web.config.